test/Microsoft.AspNetCore.Mvc.Core.Test/ConsumesAttributeTests.cs

```diff
@@ -293,6 +295,193 @@ public void Accept_ForNoRequestType_ReturnsTrueForAllConstraints(string contentT
+            Assert.False(constraint1.Accept(context));
+            context.CurrentCandidate = context.Candidates[1];
+            Assert.False(constraint2.Accept(context));
+        }
+
+        [Theory]
+        [InlineData("application/xml")]
+        [InlineData("application/custom")]
+        [InlineData("invalid/invalid")]
+        public void EndpointConstraint_Accept_UnrecognizedMediaType_SelectsTheCandidateWithoutConstraintIfPresent(string contentType)
+        {
+            // Arrange
+            var endpointWithoutConstraint = CreateEndpoint();
+            var constraint1 = new ConsumesAttribute("application/json");
+            var endpointWithConstraint = CreateEndpoint(constraint1);
+
+            var constraint2 = new ConsumesAttribute("text/xml");
+            var endpointWithConstraint2 = CreateEndpoint(constraint2);
+
+            var context = new EndpointConstraintContext();
+            context.Candidates = new List<EndpointSelectorCandidate>()
+            {
+                new EndpointSelectorCandidate(endpointWithConstraint, new [] { constraint1 }),
+                new EndpointSelectorCandidate(endpointWithConstraint2, new [] { constraint2 }),
+                new EndpointSelectorCandidate(endpointWithoutConstraint, new List<IEndpointConstraint>()),
+            };
+
+            context.HttpContext = CreateHttpContext(contentType: contentType);
+
+            // Act & Assert
+            context.CurrentCandidate = context.Candidates[0];
+            Assert.False(constraint1.Accept(context));
+
+            context.CurrentCandidate = context.Candidates[1];
+            Assert.False(constraint2.Accept(context));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        public void EndpointConstraint_Accept_ForNoRequestType_ReturnsTrueForAllConstraints(string contentType)
+        {
+            // Arrange
+            var constraint1 = new ConsumesAttribute("application/json");
+            var endpointWithConstraint = CreateEndpoint(constraint1);
+
+            var constraint2 = new ConsumesAttribute("text/xml");
+            var endpointWithConstraint2 = CreateEndpoint(constraint2);
+
+            var endpointWithoutConstraint = CreateEndpoint();
+
+            var context = new EndpointConstraintContext();
+            context.Candidates = new List<EndpointSelectorCandidate>()
+            {
+                new EndpointSelectorCandidate(endpointWithConstraint, new [] { constraint1 }),
+                new EndpointSelectorCandidate(endpointWithConstraint2, new [] { constraint2 }),
+            };
+
+            context.HttpContext = CreateHttpContext(contentType: contentType);
+
+            // Act & Assert
+            context.CurrentCandidate = context.Candidates[0];
+            Assert.True(constraint1.Accept(context));
+            context.CurrentCandidate = context.Candidates[1];
+            Assert.True(constraint2.Accept(context));
+        }
```

Comment:
I'm assuming these are clones of the existing tests and not looking at them in depth

Reply:
+1 copy-paste.